### PR TITLE
Dataprovider considers webspace settings

### PR DESCRIPTION
**Pull request by trickreich**

### CODE REVIEW

File: Content/ArticleDataProvider.php [hunk 1]
```diff
@@ -199,6 +207,29 @@ public function resolveDatasource($datasource, array $propertyParameter, array $
...
 
+    /**
+     * @param array $propertyParameter
+     * @param array $options
+     *
+     * @return string|null
+     */
+    private function getWebspaceKey(array $propertyParameter, array $options)
+    {
+        if (array_key_exists('webspace', $propertyParameter)) {
```
Comment (wachterjohannes) [inline Content/ArticleDataProvider.php]: are you sure that `webspace` and `webspaceKey` in options exists?

File: Content/ArticleDataProvider.php [hunk 2]
```diff
@@ -29,6 +29,7 @@
 use Sulu\Component\SmartContent\DataProviderAliasInterface;
 use Sulu\Component\SmartContent\DataProviderInterface;
 use Sulu\Component\SmartContent\DataProviderResult;
+use ZendSearch\Lucene\Index\Term;
```
Comment (wachterjohannes) [inline Content/ArticleDataProvider.php]: shouldnt be used in this class

File: Document/Resolver/WebspaceResolver.php [hunk 3]
```diff
@@ -54,6 +68,20 @@ public function resolveMainWebspace(WebspaceBehavior $document)
      */
     public function resolveAdditionalWebspaces(WebspaceBehavior $document)
     {
+        if (!$this->hasMoreThanOneWebspace()) {
+            return null;
```
Comment (wachterjohannes) [inline Document/Resolver/WebspaceResolver.php]: shouldnt be empty array?

File: Tests/app/config/config.yml [hunk 4]
```diff
@@ -38,3 +38,6 @@ sulu_route:
             options:
                 route_schema: "/{translator.trans(\"page\")}-{object.getPageNumber()}"
                 parent: "{object.getParent().getRoutePath()}"
+
+sulu_article:
+    default_main_webspace: 'sulu_io'
```
Comment (wachterjohannes) [inline Tests/app/config/config.yml]: add empty line

File: Content/ArticleDataProvider.php (hunk 1 above)
Comment (wachterjohannes) [reply Content/ArticleDataProvider.php]: as i can see in `vendor/sulu/sulu/src/Sulu/Component/SmartContent/ContentType.php` there is only `webspaceKey`.

**Review by wachterjohannes — CHANGES_REQUESTED** [review]

File: Document/Resolver/WebspaceResolver.php [hunk 5]
```diff
@@ -54,6 +68,20 @@ public function resolveMainWebspace(WebspaceBehavior $document)
...
+        }
+
         return $document->getAdditionalWebspaces() ? $document->getAdditionalWebspaces(): $this->defaultAdditionalWebspaces;
```
Comment (trickreich) [inline Document/Resolver/WebspaceResolver.php]: space

File: Content/ArticleDataProvider.php [hunk 6]
```diff
@@ -199,6 +206,29 @@ public function resolveDatasource($datasource, array $propertyParameter, array $
...
+     *
+     * @return string|null
+     */
+    private function getWebspaceKey(array $propertyParameter, array $options)
+    {
+        if (array_key_exists('ignore_webspaces', $propertyParameter)) {
+            $value = $propertyParameter['ignore_webspaces']->getValue();
+
+            if (true === $value) {
+                return false;
```
Comment (wachterjohannes) [inline Content/ArticleDataProvider.php]: `return null`

**Pull request merged**